### PR TITLE
Add support for set_formspec_prepend in main menu

### PR DESCRIPTION
**Pull request by rubenwardy**
This makes it easier to theme in the main menu

### CODE REVIEW

File: src/script/lua_api/l_mainmenu.cpp [hunk 1]
```diff
@@ -106,6 +106,21 @@ int ModApiMainMenu::l_update_formspec(lua_State *L)
 	return 0;
 }
 
+/******************************************************************************/
+int ModApiMainMenu::l_set_formspec_prepend(lua_State *L)
+{
+	GUIEngine* engine = getGuiEngine(L);
```
Comment (SmallJoker) [inline src/script/lua_api/l_mainmenu.cpp]: Misplaced pointer.

File: src/script/lua_api/l_mainmenu.h [hunk 2]
```diff
@@ -104,6 +104,8 @@ class ModApiMainMenu: public ModApiBase
 
 	static int l_update_formspec(lua_State *L);
 
+	static int l_set_formspec_prepend(lua_State *L);
```
Comment (SmallJoker) [inline src/script/lua_api/l_mainmenu.h]: Needs documentation in `menu_lua_api`

File: doc/menu_lua_api.txt [hunk 3]
```diff
@@ -79,6 +79,8 @@ core.explode_table_event(string) -> table
 core.explode_textlist_event(string) -> table
 ^ returns e.g. {type="CHG", index=1}
 ^ type: "INV" (no row selected), "CHG" (selected) or "DCL" (double-click)
+core.set_formspec_prepend(formspec)
+^ string to be added to every formspec, to be used for theming.
```
Comment (DS-Minetest) [inline doc/menu_lua_api.txt]: This prepend is not added to ingame fromspecs, is it?
```suggestion
^ string to be added to every mainmenu formspec, to be used for theming.
```

Comment (rubenwardy) [reply doc/menu_lua_api.txt]: lol, I'll clarify that

**Comment by paramat** [comment]
Can this solve #7060 ?

**Comment by DS-Minetest** [comment]
> Can this solve #7060 ?

No, formspec prepends can't do more than additional elements to the formspec string.

**Pull request merged**